### ISSUE TITLE
GraphQL Aliases on junction collection breaking the returned value

### ISSUE THREAD

**Issue by DamienMrtl** [directus/directus#8402]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

When using alias in a graphqQL query on a junction table the returned value of the field using the alias is null. When I remove the alias from the query the data is there.

### To Reproduce

1. Two collections with a m2m relation to get a junction table.
2. Use graphQL to query nested data from both collections.
3. Add an alias on the junction table field in the graphQL query.

### What version of Directus are you using?

9.0.0-rc95

### What version of Node.js are you using?

14.17.5

### What database are you using?

Postgres 13.2

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying Directus?

Jelastic

**Comment by azrikahar** [comment]
Can confirm this bug is present, and seems like #8332 was correct in that aliases are broken since rc.94 (even though we couldn't replicate it back then).

Managed to narrow it down to this bug being introduced in #8009's `api/src/services/graphql.ts` as reverting that specific file only makes aliases work again.

**Comment by rijkvanzanten** [comment]
@hueindahaus 

[**Unfortunately, we can not provide estimates on features or fixes**](https://docs.directus.io/getting-started/support.html#can-you-give-an-eta-for-this-feature-fix). As of now, the only way to get an accurate timeline or release date is to [sponsor](https://docs.directus.io/getting-started/support.html#sponsored-work) this task. We recognize that this may be an important feature/fix, but we are a small open-source organization with a lot to triage and complete. If you can't sponsor this ticket, then the next best thing is to increase its priority by giving it a 👍 . Or contribute a pull-request... after all, this is OSS!

---

_This is an automated response._

**Comment by filipproch** [comment]
I managed to narrow it down to the line (api/src/services/graphql.ts)
1378 `current = selection.alias?.value ?? selection.name.value;`

removing the part
`selection.alias?.value ?? `
fixes the issue

**Comment by rijkvanzanten** [comment]
Wanna open a PR with that fix @filipproch? 🙂

**Comment by jaycammarano** [comment]
> We've just started an effort to setup test cases for more and more bits of the API, but I don't think the GraphQL endpoint / services have something spun up yet, right @jaycammarano?

No. That was on my notes to discuss. Was thinking of probably doing e2e for REST and integration for GraphQL? Just so there's not a ton of repetition in the e2e tests.

**Comment by filipproch** [comment]
Ok will send PR with just the fix then

**Comment by iksent** [comment]
I can confirm this bug exists in RC100.

```
Cannot destructure property 'type' of 'field' as it is undefined.
```

My code:
```
query siteMapContests {
  contest(limit: 100) {
    s: slug # <<< Works OK without renaming "s: "
  }
}
```

**Comment by rijkvanzanten** [comment]
A new issue would be most helpful @iksent 👍🏻

**Comment by filipproch** [comment]
@iksent  And feel free to mention me, I plan to submit a PR this week

**Comment by rijkvanzanten** [comment]
Awesome, thank you!
